File: backend/app/schemas/class_.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ClassBase(BaseModel):
    grade_id: int
    name: str
    is_active: bool = True


class ClassCreate(BaseModel):
    grade_id: int
    name: str


class ClassRead(ClassBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClassUpdate(BaseModel):
    grade_id: int | None = None
    name: str | None = None
    is_active: bool | None = None

File: frontend/src/types/class.ts
// Mirrors backend/app/schemas/class_.py -- every field on
// ClassRead/ClassCreate/ClassUpdate, not just what the read-only Phase A
// table renders (see docs/FRONTEND_REQUIREMENTS.md Task 27 appendix).

export interface Class {
  id: number;
  grade_id: number;
  name: string;
  is_active: boolean;
  homeroom_room_id: number | null;
  created_at: string;
  updated_at: string;
}

export interface ClassCreate {
  grade_id: number;
  name: string;
  homeroom_room_id?: number | null;
}

export interface ClassUpdate {
  grade_id?: number;
  name?: string;
  is_active?: boolean;
  homeroom_room_id?: number | null;
}
